### PR TITLE
Node.js Homework week2

### PR DESCRIPTION
**Pull request by WadeeaKiwan**

### CODE REVIEW

File: week2/homework/src/actions.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  remove: index => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      if (index > 0 && index <= todoList.length) {
+        todoList.splice(index - 1, 1);
+        console.log('Todo item has been removed');
```
Comment (eidosam) [inline week2/homework/src/actions.js]: This message should be moved to the callback, so you don't report the success early.

File: week2/homework/src/actions.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  update: (index, updatedItem) => {
+    fs.readFile(PATH, ENCODING, (err, todoList) => {
+      if (err) {
+        if (err.code === 'ENOENT') console.log('no data found');
+        throw err;
+      }
+      todoList = JSON.parse(todoList);
+      if (index > 0 && index <= todoList.length) {
+        todoList.splice(index - 1, 1, updatedItem);
+        console.log('Todo item has been modified');
```
Comment (eidosam) [inline week2/homework/src/actions.js]: This also has to be moved to inside the callback of `fs.writeFile`.

**Comment by WadeeaKiwan** [comment]
> Very well done Wadeea! Very neat code!

Manay thanks @eidosam for your feedback.

File: week2/homework/src/actions.js (hunk 1 above)
Comment (WadeeaKiwan) [reply week2/homework/src/actions.js]: Thanks... done 👍

File: week2/homework/src/actions.js (hunk 2 above)
Comment (WadeeaKiwan) [reply week2/homework/src/actions.js]: Thanks... done 👍

**Comment by wouterkleijn** [comment]
Thanks @eidosam for the comments. Super, @WadeeaKiwan , great job!

**Pull request closed without merging**